Documentation/custom-scopes-claims-clients.md

```diff
@@ -0,0 +1,69 @@
+# Custom scopes, claims and client features
+
+This document describes the set of OAuth2 and OpenID Connect features implemented by dex.
+
+## Scopes
+
+The following is the exhaustive list of scopes supported by dex:
+
+| Name | Description |
+| ---- | ------------|
+| `openid` | Required scope for all login requests. |
+| `email` | ID token claims should include the end user's email and if that email was verified by an upstream provider. |
+| `profile` | ID token claims should include the username of the end user. |
+| `groups` | ID token claims should include a list of groups the end user is a member of. |
+| `offline_access` | Token response should include a refresh token. |
+| `audience:server:client_id:( client-id )` | Dynamic scope indicating that the ID token should be issued on behalf of another client. See the _"Cross-client trust and authorized party"_ section below. |
+
+## Custom claims
+
+Beyond the [required OpenID Connect claims][core-claims], and a handful of [standard claims][standard-claims], dex implements the following non-standard claims.
+
+| Name | Description |
+| ---- | ------------|
+| `groups` | A list of strings a user is a member of. |
```

Comment:
consider rewriting "groups: A list of strings a user is a member of" as a user can't be a member of strings. "Groups are the hierarchical representation of ....